plutus-tx/src/PlutusTx/SortedMap.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE ViewPatterns       #-}
+
+{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}
+{-# OPTIONS_GHC -fno-spec-constr #-}
+{-# OPTIONS_GHC -fno-specialise #-}
+
+module PlutusTx.SortedMap
+    ( SortedMap (..)
+    , empty
+    , singleton
+    , unsafeFromListUnique
+    , eqWith
+    ) where
+
+import Prelude qualified as Haskell
+
+import PlutusTx.Base
+import PlutusTx.Ord
+import PlutusTx.Prelude
+
+import Data.Coerce (coerce)
+
+-- | A map from @k@ to @v@ backed by a list that is supposed to be sorted from lowest to highest
+-- @k@s with no key duplicated.
+newtype SortedMap k v = UnsafeSortedMap
+    { unSortedMap :: [(k, v)]
+    } deriving stock (Haskell.Show)
+
+{-# INLINABLE empty #-}
+empty :: SortedMap k v
+empty = UnsafeSortedMap []
+
+{-# INLINE singleton #-}
+singleton :: k -> v -> SortedMap k [v]
+singleton ~k ~v = UnsafeSortedMap [(k, [v])]
```

Comment:
Why the arguments to this function are lazy? Also aren't we supposed to use Strict nowadays to the whole of plutustx prelude?